### ISSUE TITLE
Create Method for Saving and Loading Presets

### ISSUE THREAD

**Issue by RandomationGames** [RandomationGames/Randomation-Vehicle-Physics#13]
It is not easy to transfer certain properties between different instances of the same components or implement a vehicle customization system in a game. Being able to save and load presets for different components such as suspensions, transmissions, engines, wheels, and other parts would make it much easier to manage a project with many vehicles and allow them to be customized.

ScriptableObjects could potentially work for serializing properties. Perhaps customizable classes could use an interface implementing methods for saving and loading presets. Some components might need multiple groups of customizable properties, such as a wheel having a group of properties for tire friction and another one for tire size or audio clips. Users might want to maintain the same friction preset for wheels on vehicles, but load different presets for the sizes of the wheels and vice versa.

**Comment by tanakasama** [comment]
Actually, for setup a new vehicle, its quite difficult to setup (like setup animation curve ... ). If have loading presets, would be great to setup.

**Comment by tanakasama** [comment]
Hi RandomationsGames,

How about with this issues, i have problems when create new vehicles, i can config correct the animation curves variables.

**Comment by RandomationGames** [comment]
Unfortunately I don't have time to work on issues labeled as ambitious. For your project, maybe you could look into using ScriptableObjects to create your own preset system. It shouldn't be too much work to get basic system working where you can click a button in the inspector to copy all variables from a script to the ScriptableObject to save them, and another button to copy all variables from the ScriptableObject to the actual script for loading. You can look at the existing editor scripts for examples of inspector buttons.

If I was to commit this feature to the main branch, I would have to spend more time designing and testing it since many people would be using it. For one person's project you can often afford to use a less elegant and robust solution since it only has to work for your case, which is why maybe you would be able to figure out a basic solution. (I'm not sure how experienced you are with programming.)

**Comment by JustInvoke** [comment]
Yes, my other [asset](https://assetstore.unity.com/packages/templates/systems/powerslide-kart-physics-154032) uses ScriptableObjects for presets but I'm not willing to put in the time to refactor this one to work similarly. You are free to give it a shot if you would like.

**Comment by cholushkin** [comment]
no no, I am talking about saving preset feature of unity which works for any monobehaviour. The idea is that it could be more convinient than using ScriptableObjects. You will see all options right away in the inspector instead of switching to ScriptableObjects all the time

**Comment by cholushkin** [comment]
![image](https://user-images.githubusercontent.com/6879641/167400976-2a096f5d-c5d8-483f-8d44-7a7161ae91dd.png)
